### PR TITLE
NOISSUE  - Add connect and Assign members to group - UI

### PR DESCRIPTION
**Pull request by fbugarski**
### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: ui/service.go [hunk 1]
```diff
@@ -336,6 +468,21 @@ func (gs *uiService) ViewGroup(ctx context.Context, token, id string) ([]byte, e
 	return btpl.Bytes(), nil
 }
 
+func (gs *uiService) Assign(ctx context.Context, token string, groupID, groupType string, memberIDs ...string) ([]byte, error) {
+	if err := gs.sdk.Assign(memberIDs, groupType, groupID, token); err != nil {
+		return []byte{}, err
+	}
+
```
Comment (drasko) [inline ui/service.go]: Here you left an empty line, but in the function below you did not. Please choose one or another and stay consistent.

**Review by drasko — CHANGES_REQUESTED** [review]
Very good, just one small remark regarding the style. PLease fix this and we can merge.

File: ui/api/endpoint.go [hunk 2]
```diff
@@ -224,6 +224,96 @@ func removeChannelEndpoint(svc ui.Service) endpoint.Endpoint {
...
+		if err != nil {
+			return nil, err
+		}
+		return uiRes{
+			html: res,
+		}, err
+	}
+}
+
+func connectEndpoint(svc ui.Service) endpoint.Endpoint {
```
Comment (drasko) [inline ui/api/endpoint.go]: What's the difference of `connectEndpoint` and other 2 functions?

File: ui/api/transport.go [hunk 3]
```diff
@@ -300,6 +466,65 @@ func decodeGroupUpdate(_ context.Context, r *http.Request) (interface{}, error)
...
+func parseSubtopic(subtopic string) (string, error) {
+	if subtopic == "" {
+		return subtopic, nil
+	}
+
+	subtopic, err := url.QueryUnescape(subtopic)
+	if err != nil {
+		return "", errMalformedSubtopic
+	}
+	subtopic = strings.Replace(subtopic, "/", ".", -1)
```
Comment (drasko) [inline ui/api/transport.go]: Why this replacement with `.`? Can't you split on `/`?

**Review by drasko — CHANGES_REQUESTED** [review]

File: ui/api/endpoint.go (hunk 2 above)
Comment (fbugarski) [reply ui/api/endpoint.go]: `connectEndpoint` is for listing all thing connections (button connections), `connectChannelEndpoint` is for listing all channel connections, and `connectThingEndpoint` is for connecting thing to channel and opposite.

File: ui/api/transport.go (hunk 3 above)
Comment (fbugarski) [reply ui/api/transport.go]: UI doesn't support subtopic for now, i will remove this function.

File: ui/api/endpoint.go (hunk 2 above)
Comment (drasko) [reply ui/api/endpoint.go]: Then the naming is not really good. Should be `listConnections` or something, as we use for Users and Things - AFAIK we are using `list-` prefix, right?

Comment (drasko) [reply ui/api/endpoint.go:245]: Accordingly, structs in `req` and `resp` should be renamed.

Comment (drasko) [reply ui/api/endpoint.go]: Also, why are we using `svc.ViewConnections()` here - should it not be `svc.ListConnections()` according to our standard, similar to [this](https://github.com/mainflux/mainflux/blob/master/things/api/things/http/endpoint.go#L173) (and same for `viewResourceReq`)

**Review by drasko — CHANGES_REQUESTED** [review]

File: ui/api/transport.go [hunk 4]
```diff
@@ -26,13 +28,25 @@ import (
 const (
 	contentType = "text/html"
 	staticDir   = "ui/web/static"
-	token       = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MzkwODg1MTUsImlhdCI6MTYzOTA1MjUxNSwiaXNzIjoibWFpbmZsdXguYXV0aCIsInN1YiI6ImZscDFAZW1haWwuY29tIiwiaXNzdWVyX2lkIjoiYzkzY2FmYjMtYjNhNy00ZTdmLWE0NzAtMTVjMTRkOGVkMWUwIiwidHlwZSI6MH0.Ck9r2eJc2gieUdGy1Lt1BYLpYmf5rQT-KX1bckmwq28"
+	token       = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDE1MTkyNjEsImlhdCI6MTY0MTQ4MzI2MSwiaXNzIjoibWFpbmZsdXguYXV0aCIsInN1YiI6ImZscDFAZW1haWwuY29tIiwiaXNzdWVyX2lkIjoiYzkzY2FmYjMtYjNhNy00ZTdmLWE0NzAtMTVjMTRkOGVkMWUwIiwidHlwZSI6MH0.cqDOZdqiH9sXd1yuDwsv6-Mtb6_nVe_4c6cJK-iJ-Ig"
```
Comment (drasko) [inline ui/api/transport.go:32]: Leave a `TODO` comment above this `token` that it is a temporary token and will be removed once auth proxy is in place.

File: ui/service.go [hunk 5]
```diff
@@ -271,6 +282,112 @@ func (gs *uiService) RemoveChannel(ctx context.Context, token, id string) ([]byt
...
+	var btpl bytes.Buffer
+	if err := tpl.ExecuteTemplate(&btpl, "connections", data); err != nil {
+		println(err.Error())
+	}
+	return btpl.Bytes(), nil
+}
+
+func (gs *uiService) ListChannelConnections(ctx context.Context, token, id string) ([]byte, error) {
+	tpl, err := parseTemplate("connectionsttc", "connectionsttc.html")
```
Comment (drasko) [inline ui/service.go]: Please rename this file, because it is not self-documenting

**Review by drasko — CHANGES_REQUESTED** [review]

File: ui/api/transport.go (hunk 4 above)
Comment (drasko) [reply ui/api/transport.go:32]: Please add this `TODO` comment on a line in front of the token, as explained

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**